### ISSUE TITLE
Popping oscript frame does not advance prev frame

### ISSUE THREAD

**Issue by jefforkin** [giantotter/giantotter_public#161]
We can spawn model OK/Cancel dialogs on the main thread using PushScript().  However, when the script ends, it lingers until the next AdvanceEvent is called to clean it up.  This breaks the script that called it, if it is waiting on an event.

For example, if a tutorial step is waiting for someone to click the microwave, and something does a PushScript for a modal dialog and Cancels, Using the microwave will lead to popping the dialog script, rather than moving onto the next event in the waitress events script for the tutorial.

I tried to fix this by having AdvanceEvent continue advancing in the calling script after a pop, but this broke other things.

**Comment by jefforkin** [comment]
no longer relevant
